core/ecschema-metadata/src/UnitConversion/Parser.ts

```diff
@@ -0,0 +1,43 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+const expressionRgx = /^(([A-Z]\w*:)?([A-Z]\w*|\[([A-Z]\w*:)?[A-Z]\w*\])(\(-?\d+\))?(\*(?!$)|$))+$/i;
+const tokenRgx = /(?:(\[)?((?:[A-Z]\w*:)?[A-Z]\w*)\]?)(?:\((-?\d+)\))?/i;
+const sp = "*";
+
+enum Tokens {
+  Bracket = 1,
+  Word = 2,
+  Exponent = 3,
+}
+
+export interface Definition {
```

Comment:
I would say the map of all the 'Definition' objects is really the definition.  So we need to come up with a better name for this.  We called them Symbols in the C++ implementation ... but I'm open to other suggestions ... and well we called the parsed definition the 'expression' ... but I'm not sure I'm in love with the old names.

Reply:
Maybe PartialDefinition or DefinitionFragment or DefinitionSegment because each is a part of the whole definition.